### PR TITLE
set timeout for grpc requests

### PR DESCRIPTION
**Pull request by andrenerd**

### CODE REVIEW

**Comment by andrenerd** [comment]
@shravanshetty1
Shravan, are the tests consistent?
There is a test error. I'm not sure it's related to the update.

And the same error was in a pull request you merged recently:
https://github.com/0chain/blobber/pull/73/checks

**Comment by guruhubb** [comment]
@Sriep please review this

**Comment by andrenerd** [comment]
@shravanshetty1 
seems the fixed the underlying issue with tests:
https://github.com/0chain/blobber/commit/84c35287f487041131ae3550d5d7a36ce957b0da

File: code/go/0chain.net/blobbercore/handler/grpchandler.go [hunk 1]
```diff
@@ -52,6 +57,14 @@ func unaryDatabaseTransactionInjector() grpc.UnaryServerInterceptor {
 	}
 }
 
+func unaryTimeoutInterceptor() grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		deadline := time.Now().Add(time.Duration(TIMEOUT_SECONDS * time.Second))
+		ctx, _ = context.WithDeadline(ctx, deadline)
+		return handler(ctx, req)
```
Comment (shravanshetty1) [inline code/go/0chain.net/blobbercore/handler/grpchandler.go:64]: This should return request timeout error when deadline has exceeded, by checking the done channel.

**Review by shravanshetty1 — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/blobbercore/handler/grpchandler.go (hunk 1 above)
Comment (andrenerd) [reply code/go/0chain.net/blobbercore/handler/grpchandler.go:64]: @shravanshetty1 
it does. it already does.

context lib and grpc lib do that together, no custom code to create and pass error is needed there.
context checks for deadline and send cancel signal up, grpc transforms the signal to specific error and send it back as one of the handler returned param. all the "outer" handlers also get the error.

specifically, handler returns and clients receive such error:
```
rpc error: code = DeadlineExceeded desc = context deadline exceeded
```

**Comment by andrenerd** [comment]
@shravanshetty1 please, check it again. if it's ok, lets merge.

File: code/go/0chain.net/blobbercore/handler/grpchandler.go (hunk 1 above)
Comment (shravanshetty1) [reply code/go/0chain.net/blobbercore/handler/grpchandler.go:64]: https://grpc.io/blog/deadlines/

According to this article they do handle deadlines, but couldnt find the code for it. Will test if it works and then approve.

Comment (andrenerd) [reply code/go/0chain.net/blobbercore/handler/grpchandler.go:64]: @shravanshetty1 
if no other changes needed, please, mark the review accordingly. thanks.

**Review by Sriep — APPROVED** [review]

**Pull request merged**